bc_obps/service/operation_service.py

```diff
@@ -0,0 +1,249 @@
+                    "last_name": payload.external_point_of_contact_last_name,
+                    "position_title": payload.external_point_of_contact_position_title,
+                    "email": payload.external_point_of_contact_email,
+                    "phone_number": payload.external_point_of_contact_phone_number,
+                }
+
+                external_poc = ContactDataAccessService.update_or_create(
+                    point_of_contact_id, external_contact_data, user_guid
+                )
+                operation.point_of_contact = external_poc
+            operation.save(update_fields=['point_of_contact'])
+
+        elif form_section == 3 and payload.statutory_declaration:
+            existing_statutory_document: Document = DocumentService.get_existing_statutory_declaration_by_operation_id(
+                operation_id
+            )
+            # if there is an existing statutory declaration document, check if the new one is different
+            if existing_statutory_document:
+                # We need to check if the file has changed, if it has, we need to delete the old one and create a new one
+                if not files_have_same_hash(payload.statutory_declaration, existing_statutory_document.file):
+                    existing_statutory_document.delete()
+                    DocumentDataAccessService.create_statutory_declaration(
+                        user_guid, payload.statutory_declaration, operation
+                    )
+
+            else:
+                # if there is no existing statutory declaration document, create a new one
+                DocumentDataAccessService.create_statutory_declaration(
+                    user_guid, payload.statutory_declaration, operation
+                )
+
+        if submit == "true":
+            """
+            if the PUT request has submit == "true" (i.e., user has clicked Submit button in UI form), the desired behaviour depends on
+            the Operation's status:
+                - if operation.status was already "Approved", it should remain Approved and the submission date should not be altered
+                - if operation.status was "Changes Requested", it should switch to Pending
+                - if operation.status was "Declined", it should switch to Pending
+                - if operation.status was "Not Started", it should switch to Pending
+                - if operation.status was "Pending", it should remain as Pending
+            """
+            if operation.status != Operation.Statuses.APPROVED:
+                operation.status = Operation.Statuses.PENDING
+                operation.submission_date = datetime.now(pytz.utc)
+                operation.save(update_fields=['status', 'submission_date'])
+        operation.set_create_or_update(user_guid)
+        return operation
+
+    @classmethod
+    def list_operations(cls, user_guid, filters: OperationFilterSchema = Query(...)):
```

Comment:
I was torn on this one because some of the stuff in here seems more service-y than purely db access. I moved the query into the db layer but left some of the table filtering/sorting in here, let me know what you think

Reply:
Makes sense to me! Just a suggestion for a good refactor; In the following code, there is only one difference and that is using `OperationDataAccessService.get_all_operations_for_irc_user()` or `OperationDataAccessService.get_all_operations_for_industry_user(user)` based on the user role.

![Screenshot 2024-05-02 at 8 41 24 AM](https://github.com/bcgov/cas-registration/assets/52381975/50008207-20e7-4947-9aee-ad979c6c3641)

We might be able to do something like this (⚠️ NOT TESTED):
```python
    base_qs = OperationDataAccessService.get_all_operations_for_user(user)
    filters = [
        Q(bcghg_id__icontains=bcghg_id) if bcghg_id else Q(),
        Q(bc_obps_regulated_operation__id__icontains=bc_obps_regulated_operation)
        if bc_obps_regulated_operation
        else Q(),
        Q(name__icontains=name) if name else Q(),
        Q(operator__legal_name__icontains=operator) if operator else Q(),
        Q(status__icontains=status) if status else Q(),
    ]

    qs = base_qs.filter(*filters).order_by(f"{sort_direction}{sort_field}")
```

And then we can combine `get_all_operations_for_irc_user` and `get_all_operations_for_industry_user` to return operation manager based on the user role.